esphome/components/sensor/filter.cpp

```
@@ -187,6 +187,29 @@ optional<float> ExponentialMovingAverageFilter::new_value(float value) {
 void ExponentialMovingAverageFilter::set_send_every(size_t send_every) { this->send_every_ = send_every; }
 void ExponentialMovingAverageFilter::set_alpha(float alpha) { this->alpha_ = alpha; }
 
+// PeriodicalAverageFilter
+PeriodicalAverageFilter::PeriodicalAverageFilter(uint32_t time_period) : time_period_(time_period) {}
+
+optional<float> PeriodicalAverageFilter::new_value(float value) {
+  ESP_LOGVV(TAG, "PeriodicalAverageFilter(%p)::new_value(value=%f)", this, value);
+  this->sum_ += value;
+  this->n_++;
+
+  return {};
+}
+void PeriodicalAverageFilter::setup() {
+  this->set_interval("periodical_average", this->time_period_, [this]() {
+    ESP_LOGVV(TAG, "PeriodicalAverageFilter(%p)::interval(sum=%f, n=%i)", this, this->sum_, this->n_);
+    if (this->n_ == 0)
```

If no sample occured in that time interval I think we should not silently drop that.

IMO, Let's send NAN in that case (if the user does want that dropped, then they can use the filter out nan filter)

---

Seem right. @rmeissn we should specify it in the documentation.

---

@OttoWinter the change is done. But I asked myself, What happen in Home Assistant when the filter sends NaN value? Does the HA sensor become unavailable?